src/modules/measure/measure-service.js

```diff
@@ -0,0 +1,222 @@
+
+        function formatLineResult(geometry, projection, geodesic) {
+            var length = calculateLength(geometry, projection, geodesic);
+            var output;
+            if (length > 1000) {
+                output = anol.helper.round((length / 1000), 3) + ' ' + 'km';
+            } else {
+                output = anol.helper.round(length, 2) + ' ' + 'm';
+            }
+            output = output.replace('.', ',')
+            return output;
+        };
+
+        function formatAreaResult(geometry, projection, geodesic, addCircumference) {
+            var area = calculateArea(geometry, projection, geodesic);
+            if (area === 0) {
+                return '';
+            }
+
+            var output;
+            if (area > 100000) {
+                output = anol.helper.round((area / 10000), 3) +
+                    ' ' + 'ha';
+            } else {
+                output = anol.helper.round(area, 2) +
+                    ' ' + 'm<sup>2</sup>';
+            }
+            output = output.replace('.', ',')
+
+            if (addCircumference) {
+                // calculate line length for area and add to label
+                var length = 0;
+                var coordinates = geometry.getCoordinates();
+                var coords = coordinates[0];
+                angular.forEach(coords, function (coord, idx) {
+                    if (idx !== coords.length - 1) {
+                        length += calculateLength(new LineString([coord, coords[idx + 1]]));
+                    }
+                })
+                if (length > 1000) {
+                    output += '<br>' + anol.helper.round((length / 1000), 3) + ' ' + 'km';
+                } else {
+                    output += '<br>' + anol.helper.round(length, 2) + ' ' + 'm';
+                }
+                output = output.replace('.', ',')
+            }
+
+            return output;
+        };
+
+        var measureStyle = function (feature, labelSegments) {
+            var geometry = feature.getGeometry();
+            var styles = [
+                new Style({
+                    fill: new Fill({
+                        color: 'rgba(255, 255, 255, 0.2)'
+                    }),
+                    stroke: new Stroke({
+                        color: 'rgba(0, 0, 0, 0.5)',
+                        lineDash: [10, 10],
+                        width: 2,
+                        opacity: 0.5
+                    }),
+                    image: new CircleStyle({
+                        radius: 5,
+                        stroke: new Stroke({
+                            color: 'rgba(0, 0, 0, 0.7)'
+                        })
+                    })
+                })
+            ];
+
+            function lengthAsString(start, end) {
+                var geometry = new LineString([start, end])
+                var projection = MapService.getMap().getView().getProjection();
+
+                var length;
+                // TODO load from Service
```

Comment:
minor: is this still a TODO?

Reply:
That's a good question. I copied the functions over from measure-directive.js. I am not 100% sure what it refers to, but to me it seems to be the geodesic flag (below the TODO), not the length. The code fragment currently ignores the possibility to perform geodesic measurements (it is always set to false).

My suggestion is to check the code paths for geodesic measurements as soon as we need them. I will add a ticket.